mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -49,6 +51,8 @@ pub(crate) mod put_object;
 
 #[derive(Debug, Clone, Default)]
 pub struct S3ClientConfig {
+    pub profile_name_override: Option<String>,
```

Comment:
Aren't these options mutually exclusive? Shall we enforce it here and in CLI options?

Reply:
They are basically optional, if customer omit some of them, those configuration should never reflect to client

Reply:
They're not quite mutually exclusive -- profiles can contain things other than credentials, like CLI configuration. We don't use any of those today, but conceivably could in the future.